### PR TITLE
[bug-fix] Fixes un-deploying an API deployed to multiple labels form APICTL

### PR DESCRIPTION
**Pull request by slahirucd7**
### Purpose
Fixes error when un-deploying an API deployed to multiple labels through APICTL.
image

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2363

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by pubudu538 — APPROVED** [review]

**Comment by VirajSalaka** [comment]
Would be better if we can have an integration test to cover the scenario?
Something like this.
https://github.com/wso2/product-microgateway/commit/262838186350c855da83d5283189fe2883911d93

**Comment by slahirucd7** [comment]
> Would be better if we can have an integration test to cover the scenario? Something like this. [2628381](https://github.com/wso2/product-microgateway/commit/262838186350c855da83d5283189fe2883911d93)

Ack. I'll add  a test with `Add integration tests for disable-security` PR.

File: adapter/internal/discovery/xds/server.go [hunk 1]
```diff
@@ -502,8 +502,15 @@ func DeleteAPIs(vhost, apiName, version string, environments []string, organizat
...
 			}
 			deletedVhosts[vh] = void
+			
+			for val := range deletedVhosts {
+				existingLabels := orgIDOpenAPIEnvoyMap[organizationID][apiIdentifier]
+				if val == vh && len(existingLabels) == 0 {
+					logger.LoggerXds.Infof("Vhost : %v  deleted since there is no gateways assigned to it.", vh)
+					delete(apiToVhostsMap[apiNameVersionHashedID],val)
+				}
+			}
```
Comment (renuka-fernando) [inline adapter/internal/discovery/xds/server.go:512]: Can't we get this out from the parent for loop? And avoid checking `val == vh` in if condition?
Just a suggestion

Comment (slahirucd7) [reply adapter/internal/discovery/xds/server.go:512]: we need to check `val == vh` equality to determine deleting vhost properly. Also we need to do this inside the parent loop in order to delete the vhost at the time of gateway deletion (when there are no gateways allocated to the vhost vhost will also get deleted).

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20211215.32

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211215.43

**Pull request merged**